backend/App.js

```diff
@@ -1,12 +1,13 @@
+require('dotenv').config();
 const express = require('express');
-
-const app = express();
-const PORT = 4000;
 const {
-  troops, buildings, resources, notification,
+  users, troops, buildings, resources, notification, progresses,
```

Comment:
This is the progress table stores all the units including buildings and troops that are under construction or upgrading. When the front fetching the units, the back end will first check is there any finished units in the progress table. If so, it will add the finished ones in the buildings/troops table.